lib/iris/coords.py

```diff
@@ -268,78 +272,269 @@ def _has_lazy_values(self):
+        title_str = f"{self.name()} / ({units_str})"
+        cls_str = type(self).__name__
+        shape_str = str(self.shape)
+
+        # Implement conditional defaults for control args.
+        if max_values is None:
+            max_values = 5 if shorten else 15
+        precision = 3 if shorten else None
+        n_indent = 4
+        indent = " " * n_indent
+        addline = "\n" + indent
+        if linewidth is not None:
+            given_array_width = linewidth
+        else:
+            given_array_width = np.get_printoptions()["linewidth"]
+        using_array_width = given_array_width - n_indent * 2
+        # Make a printout of the main data array (or maybe not, if lazy).
+        if self._has_lazy_values():
+            data_str = "<lazy>"
+        elif max_values == 0:
+            data_str = "[...]"
+        else:
+            data_str = array_summary(
+                self._values,
+                n_max=max_values,
+                n_edge=edgeitems,
+                linewidth=using_array_width,
+                precision=precision,
+            )
+
+        if shorten:
+            result = f"<{cls_str}: {title_str}  "
+            if data_str == "<lazy>":
+                # Data summary is invalid due to being lazy : show shape.
+                data_str += f"  shape{shape_str}"
+            else:
+                # Flatten to a single line, reducing repeated spaces.
+                def flatten_array_str(array_str):
+                    array_str = array_str.replace("\n", " ")
+                    array_str = array_str.replace("\t", " ")
+                    while "  " in array_str:
+                        array_str = array_str.replace("  ", " ")
+                    return array_str
+
+                data_str = flatten_array_str(data_str)
+                # Adjust maximum-width to allow for the title width in the
+                # repr form.
+                using_array_width = given_array_width - len(result)
+                # Work out whether to include a summary of the data values
+                if len(data_str) > using_array_width:
+                    # Make one more attempt, printing just the *first* point,
+                    # as this is useful for dates.
+                    data_str = data_str = array_summary(
+                        self._values[:1],
+                        n_max=max_values,
+                        n_edge=edgeitems,
+                        linewidth=using_array_width,
+                        precision=precision,
+                    )
+                    data_str = flatten_array_str(data_str)
+                    data_str = data_str[:-1] + ", ...]"
+                    if len(data_str) > using_array_width:
+                        # Data summary is still too long : replace with array
+                        # "placeholder" representation.
+                        data_str = "[...]"
+
+                if self.has_bounds():
+                    data_str += "+bnds"
```

Comment:
Can I persuade you to change this to 
```suggestion
                    data_str += "+bounds"
```
It is only two characers after all, and I would prefer we use the official term "bounds" rather than invent a new term "bnds". Dropping the vowels may be natural and obvious to us but I would be concerned that this may trip some people up (non-native english speakers, perhaps?). 

I think it's fine to use `bnds` to name a variable in some example code but not in part of the code in this way

For two vowels I would say it is worth being explicit